salt/oozie_ssh_auth/add_to_authorized_keys.sls

```diff
@@ -0,0 +1,14 @@
+{% set cluster = salt['pnda.cluster_name']() %}
+{% set os_user = pillar['os_user'] %}
+{% set oozie_minion_hostname = salt['mine.get']('G@pnda_cluster:{} and G@roles:oozie_database'.format(cluster),
```

Comment:
Try to use pnda salt script. For example, if you just want to get oozie database host name here, use 
{% set oozie_db_host = salt['pnda.get_hosts_for_role']('oozie_database')[0] %}